arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -1085,29 +1090,18 @@ double RocksDBReplicationContext::expires() const {
   return _expires;
 }
 
-bool RocksDBReplicationContext::isDeleted() const {
-  MUTEX_LOCKER(locker, _contextLock);
-  return _isDeleted;
-}
-
-void RocksDBReplicationContext::setDeleted() {
-  MUTEX_LOCKER(locker, _contextLock);
-  _isDeleted = true;
-}
-
-bool RocksDBReplicationContext::isUsed() const {
-  MUTEX_LOCKER(locker, _contextLock);
-  return (_users > 0);
-}
+void RocksDBReplicationContext::extendLifetime(double ttl) {
+  double now = TRI_microtime();
 
-void RocksDBReplicationContext::use(double ttl) {
   MUTEX_LOCKER(locker, _contextLock);
-  TRI_ASSERT(!_isDeleted);
 
-  ++_users;
```

Comment:
why is it removed?

Reply:
We don't need hand-rolled usage counters anymore, as we managed the object via shared_ptr now.
The replication contexts manager maintains a map of these shared_ptrs. Any users of a replication context will get a shared_ptr copy from the manager. Once a user is done, the shared_ptr will go out of scope and reduce its use count.

Additionally, users can request deletion of an object from the manager, which will then simply remove it from its internal map. Or the manager will GC outdated/stale replication contexts, again by removing them from its map.
This simplifies the object handling a lot, and not deletion flag or usage counter is needed on the replication contexts anymore.

Reply:
I see